state-chain/chains/src/sol.rs

```diff
@@ -219,17 +212,30 @@ impl FeeEstimationApi<Solana> for SolTrackedData {
 				},
 		);
 
-		LAMPORTS_PER_SIGNATURE.saturating_add(
-			// It should never approach overflow but just in case
-			sp_std::cmp::min(
-				SolAmount::MAX as u128,
-				(self.priority_fee as u128 * compute_units_per_fetch as u128)
-					.div_ceil(MICROLAMPORTS_PER_LAMPORT.into()),
-			) as SolAmount,
-		)
+		calculate_gas_fee(self.priority_fee, compute_units_per_fetch)
 	}
 }
 
+// Calculate the gas fee for a transaction given its compute units and a priority fee.
+fn calculate_gas_fee(
+	priority_fee: <Solana as crate::Chain>::ChainAmount,
+	compute_units: SolComputeLimit,
+) -> <Solana as crate::Chain>::ChainAmount {
+	use compute_units_costs::*;
+
+	// Match the minimum broadcast prio fee
+	let priority_fee = sp_std::cmp::max(priority_fee, MIN_COMPUTE_PRICE);
```

Comment:
Does this mean we been estimating too low so far? (ie. this change will raise the fees?)

Reply:
yes but the priority fee is in microlamports so it's extremely low. It'd require the priority fee to go up a few orders of magnitude to make any difference. To just be on par with the base fee it'd require 2-3 orders of magnitude.

Reply:
Sure but we're raising the prio fee to MIN_COMPUTE_PRICE here. Is the MIN_COMPUTE_PRICE really low too?

Reply:
yes, extremely low. For our transactions the fee will go from 5000 lamports to 5001 lamports. Can't make any difference to any user.